### PR TITLE
Mesh_3 - use `manifold()` and `facet_min_size` together

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

`CGAL::parameters::manifold()` option of `CGAL::make_mesh_3()` enables the mesher to fix non-manifold edges and non-manifold vertices via incident facets refinement.
The facet criterion `facet_min_size` prevents Delaunay refinement from refining facets smaller than the given bound.

Until now, `manifold()` was not taking `facet_min_size` into account.
Now it does.

## Release Management

* Affected package(s): Mesh_3
* Solution to discussion #7665 
* License and copyright ownership: unchanged

### CODE REVIEW

**Comment by janetournois** [comment]
@lrineau this combination of parameters for a labeled image works in the demo

File: Mesh_3/include/CGAL/Mesh_3/Refine_facets_manifold_base.h [hunk 1]
```diff
@@ -239,7 +242,20 @@ class Refine_facets_manifold_base
...
 
-    return biggest_facet;
+    return { biggest_facet, biggest_sq_dist };
+  }
+
+  bool is_smaller_than_min_size(const FT& sq_dist) const
+  {
+    if(!m_facet_min_size)
+      return false;
+    return sq_dist < CGAL::square(*m_facet_min_size);
```
Comment (lrineau) [inline Mesh_3/include/CGAL/Mesh_3/Refine_facets_manifold_base.h:252]: As this is the only usage of the value of `m_facet_min_size`, I suggest it is renamed `m_squared_facet_min_size` and constructed directly as a square.

**Review by lrineau — COMMENTED** [review]
The PR looks good. Not that I have a remark about `squared_min_radius_bound` (see below).

And a question: it is planned to document this feature?

**Comment by janetournois** [comment]
> And a question: it is planned to document this feature?

Actually it was not mentioned in the documentation that `min_size` and `manifold` were not compatible. So now they are and it's already documented :-)

**Review by lrineau — APPROVED** [review]

**Comment by janetournois** [comment]
There is an issue with the data in the test that I did not have time to fix yet

**Comment by janetournois** [comment]
Ready to be tested again!
VS2017 Debug passes locally now

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-116

**Pull request merged**